parlai/utils/world_logging.py

```diff
@@ -72,23 +129,25 @@ def convert_to_labeled_data(self, log):
         return out
 
     def write_parlai_format(self, outfile):
-        print('[ saving log to {} ]'.format(outfile))
-        fw = open(outfile, 'w')
-        for episode in self._logs:
-            ep = self.convert_to_labeled_data(episode)
-            for a in ep:
-                txt = msg_to_str(a)
-                fw.write(txt + '\n')
-            fw.write('\n')
-        fw.close()
-
-    def write_json_format(self, outfile):
-        print('[ saving log to {} ]'.format(outfile))
+        print('[ Saving log to {} in ParlAI format ]'.format(outfile))
+        with open(outfile, 'w') as fw:
+            for episode in tqdm(self._log):
+                ep = self.convert_to_labeled_data(episode)
+                for a in ep:
+                    txt = msg_to_str(a)
+                    fw.write(txt + '\n')
+                fw.write('\n')
+
+    def write_jsonl_format(self, outfile):
+        print('[ Saving log to {} in jsonl format ]'.format(outfile))
         with open(outfile, 'w') as of:
```

Comment:
Perhaps I'm misreading but this reads like it's only going to end up containing the very last batch, because it will continuously rewrite over and over and over each call to `.parley()`

Reply:
this is actually what was happening in the old code, nothing is saved... however; in this code it should be fixed